src/components/pages/profile/[name]/Profile.tsx

```diff
@@ -177,13 +177,16 @@ const ProfileContent = ({ isSelf, isLoading: parentIsLoading, name }: Props) =>
   // profile.decryptedName fetches labels from NW/subgraph
   // normalisedName fetches labels from localStorage
   useEffect(() => {
+    if (!tab || !tabs.includes(tab)) {
+      router.replace(`/${normalisedName}`)
+    }
     shouldRedirect(router, 'Profile.tsx', '/profile', {
```

Comment:
Can you move the redirect logic into `shouldRedirect.tx`?  That way all redirect logic is in the same area.